### DISCUSSION TITLE
Load many videos

### DISCUSSION THREAD

**Discussion by rolandozigre** [pimcore/pimcore#16869] [Data-Hub]
Hello, 
I want to load many videos MP4 for an object in data hub, like a gallery folder. Is it possible? If yes how can I do?

**Comment by trnqeu** [comment]
I have the same problem. Can't find a solution. Seems impossibile to import more than a single media. With the previous importer it was super easy, now it's impossibile and I can't understand why.

**Comment by skoch98** [comment]
Hi, I'm not entirely sure what you're looking for, but if I understand correctly, you're trying to retrieve a list of assets based on an object relational field, right? This can be tricky because the serialization process will consume memory. You need to ensure that the server and PHP settings can handle it.

In the example below, I've created a `VideoChannel` class with an advanced many-to-many relationship to a new `VideoPlaylist` class. The `VideoPlaylist` has a simple many-to-many relationship restricted to assets of the video type.

```graphql
query {
  getVideoChannel(id: 14) {
    playlists {
      element {
      	... on object_VideoPlaylist {
          videos {
            ... on asset {
              id,
              fullpath,
              data
            }
          }
      	} 
      }
    }
  }
}
```

To keep the required memory low you can request the relations and use the asset id list to create a pagination request:

#### Request the asset id list with some meta data
```graphql
query {
  getVideoChannel(id: 14) {
    playlists {
      element {
      	... on object_VideoPlaylist {
          videos {
            ... on asset {
              id,
              fullpath,
              duration
            }
          }
      	} 
      }
    }
  }
}
```

#### Get asset by a single request per asset
```graphql
query {
  getAsset(id: 15) {
    data
  }
}
```

**Comment by trnqeu** [comment]
Hi.Thanks for the reply. Actually, I was trying to do this thorugh the data importer interface. What I wanted to do is create a relationship between multiple media assets and a single object through a csv importer. I want to have a column of my csv file with a list of media paths in every cell, separated by ;. What I want to get is a media gallery (video, images or pdf) related to a single object, via csv importer.

Best,

s
